workspaces/shared.mk

```diff
@@ -0,0 +1,181 @@
+# shared.mk: Shared make rules for test programs / policies
+#
+# AUTHORS
+#
+# The Veracruz Development Team.
+#
+# COPYRIGHT
+#
+# See the `LICENSE_MIT.markdown` file in the Veracruz root director for licensing
+# and copyright information.
+
+WARNING_COLOR := "\e[1;33m"
+INFO_COLOR := "\e[1;32m"
+RESET_COLOR := "\e[0m"
+PLATFORM := $(shell uname)
+
+###################################################
+# Wasm programs
+
+WASM_PROG_LIST = random-source.wasm \
+				linear-regression.wasm \
+				string-edit-distance.wasm \
+				intersection-set-sum.wasm \
+				private-set-intersection.wasm \
+				idash2017-logistic-regression.wasm \
+				moving-average-convergence-divergence.wasm \
+				private-set-intersection-sum.wasm \
+				number-stream-accumulation.wasm \
+				read-file.wasm \
+				shamir-secret-sharing.wasm
+
+
+WASM_PROG_FILES = $(patsubst %.wasm, $(OUT_DIR)/%.wasm, $(WASM_PROG_LIST))
+
+$(OUT_DIR):
+	@mkdir -p $@
+
+wasm-files: $(OUT_DIR) $(WASM_PROG_FILES)
+
+$(OUT_DIR)/%.wasm: $(WORKSPACE_DIR)/applications/target/wasm32-wasi/$(PROFILE_PATH)/%.wasm
+	cp $< $@
+
+$(WORKSPACE_DIR)/applications/target/wasm32-wasi/$(PROFILE_PATH)/%.wasm:
+	$(MAKE) -C $(WORKSPACE_DIR)/applications
+
+###################################################
+# Generate Policy Files
+
+# Numbers for wasi rights
+FD_DATASYNC             := $(shell echo "2^0"  | bc)
+FD_READ                 := $(shell echo "2^1"  | bc)
+FD_SEEK                 := $(shell echo "2^2"  | bc)
+FD_FDSTAT_SET_FLAGS     := $(shell echo "2^3"  | bc)
+FD_SYNC                 := $(shell echo "2^4"  | bc)
+FD_TELL                 := $(shell echo "2^5"  | bc)
+FD_WRITE                := $(shell echo "2^6"  | bc)
+FD_ADVISE               := $(shell echo "2^7"  | bc)
+FD_ALLOCATE             := $(shell echo "2^8"  | bc)
+PATH_CREATE_DIRECTORY   := $(shell echo "2^9"  | bc)
+PATH_CREATE_FILE        := $(shell echo "2^10" | bc)
+PATH_LINK_SOURCE        := $(shell echo "2^11" | bc)
+PATH_LINK_TARGET        := $(shell echo "2^12" | bc)
+PATH_OPEN               := $(shell echo "2^13" | bc)
+FD_READDIR              := $(shell echo "2^14" | bc)
+PATH_READLINK           := $(shell echo "2^15" | bc)
+PATH_RENAME_SOURCE      := $(shell echo "2^16" | bc)
+PATH_RENAME_TARGET      := $(shell echo "2^17" | bc)
+PATH_FILESTAT_GET       := $(shell echo "2^18" | bc)
+PATH_FILESTAT_SET_SIZE  := $(shell echo "2^19" | bc)
+PATH_FILESTAT_SET_TIMES := $(shell echo "2^20" | bc)
+FD_FILESTAT_GET         := $(shell echo "2^21" | bc)
+FD_FILESTAT_SET_SIZE    := $(shell echo "2^22" | bc)
+FD_FILESTAT_SET_TIMES   := $(shell echo "2^23" | bc)
+PATH_SYMLINK            := $(shell echo "2^24" | bc)
+PATH_REMOVE_DIRECTORY   := $(shell echo "2^25" | bc)
+PATH_UNLINK_FILE        := $(shell echo "2^26" | bc)
+POLL_FD_READWRITE       := $(shell echo "2^27" | bc)
+SOCK_SHUTDOWN           := $(shell echo "2^28" | bc)
+# Common rights
+READ_RIGHT       := $(shell echo $(FD_READ) + $(FD_SEEK) + $(PATH_OPEN) | bc)
```

Comment:
As mentioned on Slack, the definition of `READ_RIGHT` should include `$(FD_READDIR)`. If this file was forked from `test-collateral/Makefile` it would be good to check if there are any other changes that were made in the original file since then and should be carried over into this file.

Or into any other file that was copied from elsewhere, I suppose, though that could be a lot of work.

Reply:
I've pushed a commit that seems to fix some of the `make test-server` tests.

Reply:
Great!